extensions/common/node_info_cache.cc

```diff
@@ -57,7 +57,18 @@ bool getNodeInfo(StringView peer_metadata_key,
 }  // namespace
 
 NodeInfoPtr NodeInfoCache::getPeerById(StringView peer_metadata_id_key,
-                                       StringView peer_metadata_key) {
+                                       StringView peer_metadata_key,
+                                       std::string& peer_id) {
+  if (!getValue({"filter_state", peer_metadata_id_key}, &peer_id)) {
+    LOG_DEBUG(absl::StrCat("cannot get metadata for: ", peer_metadata_id_key));
+    return nullptr;
+  }
+  if (peer_id == "") {
```

Comment:
Since mx filter sets peer_id to unknown for incapable peer, when will this condition happen?

Reply:
Left from refactoring.. removed